samples/drivers/led_apa102c/src/main.c

```diff
@@ -55,19 +55,20 @@ u32_t rgb[] = {
 /* Number of LEDS linked together */
 #define NUM_LEDS	1
 
-void send_rgb(struct device *gpio_dev, u32_t rgb)
+void send_rgb(struct device *gpio_dev, u32_t rgb_val)
```

Comment:
So, is no-shadiow reach as far as not being able to shadow global vars?? But that's totally crazy, I always thought it's only for function-local vars in different scopes of the same function.

It's actually leaking abstractions and doesn't scale. Consider that you had a perfectly find source file, then #included something which randomly declares extern int foo for foo you use in your source, kaboom.

Reply:
There are several ways you can configure this warning option: https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html

> It's actually leaking abstractions and doesn't scale. Consider that you had a perfectly fine source file, then #included something which randomly declares extern int foo for foo you use in your source

How often does that happen in practice and is intentional?
I think for the core parts of the kernel this is useful to have on.

Reply:
> There are several ways you can configure this warning option: https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html

Good to know, thanks. Hence, I'm finding `-Wshadow=local` to have the least user surprise effect.

> How often does that happen in practice and is intentional?

I find it useful to be able to use names like "i", "p", "len", "size", "pkt", etc. when intentionally local, without worrying that a completely independent variable with such a name exists in upper scope. Of course, names like that are not suitable for global variable names. (And for public names, I think there should be a namespace prefix at all.)

> I think for the core parts of the kernel this is useful to have on.

I generally came here to hopefully approve this patch, but found points worth being discussed/known by wider developer audience. Hence my suggestion to post that as an RFC on mailing list, so more people were in loop, and also consider it in the context of following some well-known coding guidelines (like mentioned MISRA).